pkg/platform/abstract/project/external/leader/iguazio/synchronizer.go

```diff
@@ -130,22 +118,28 @@ func (c *Synchronizer) getModifiedProjects(leaderProjects []platform.Project, in
 	return
 }
 
-func (c *Synchronizer) synchronizeProjectsAccordingToLeader() error {
+func (c *Synchronizer) SynchronizeProjectsFromLeader() error {
 
-	// fetch projects from leader (created/updated since last sync)
-	leaderProjects, err := c.leaderClient.GetAll(c.lastSuccessfulSyncTimestamp)
+	// fetch updated projects from leader
+	leaderProjects, err := c.leaderClient.GetUpdatedAfter(c.mostRecentUpdatedProjectTime)
 	if err != nil {
-		return errors.Wrap(err, "Failed to get leader projects")
+		return errors.Wrap(err, "Failed to get projects from leader")
 	}
 
 	// fetch all internal projects
 	internalProjects, err := c.internalProjectsClient.Get(&platform.GetProjectsOptions{})
 	if err != nil {
-		return errors.Wrapf(err, "Failed to get all internal projects")
+		return errors.Wrapf(err, "Failed to get projects from internal client")
 	}
 
 	// filter modified projects
-	projectsToCreate, projectsToUpdate := c.getModifiedProjects(leaderProjects, internalProjects)
+	projectsToCreate, projectsToUpdate, mostRecentUpdatedProjectTime := c.GetModifiedProjects(leaderProjects, internalProjects)
+
+	// update most recent updated project time
+	if mostRecentUpdatedProjectTime != nil {
+		c.mostRecentUpdatedProjectTime = mostRecentUpdatedProjectTime
```

Comment:
if you simply return `mostRecentUpdatedProjectTime` and get it as arg as well, you wouldn't have to keep it as a data field (redundant state). It can just be a local variable in `startSynchronizationLoop`